README.md

```diff
@@ -440,6 +441,218 @@ int vi = jn.get<int>();
 
 // etc.
 ```
+### Arbitrary types conversions
+
+Every type can be serialized in JSON, not just STL-containers and scalar types.
+Usually, you would do something along those lines:
+
+```cpp
+namespace ns {
+struct person { std::string name; std::string address; int age; };
+}
+// convert to JSON
+json j;
+ns::person p = createSomeone();
+j["name"] = p.name;
+j["address"] = p.address;
+j["age"] = p.age;
+
+// ...
+
+// convert from JSON
+ns::person p {j["name"].get<std::string>(), j["address"].get<std::string>(), j["age"].get<int>()};
+```
+
+It works, but that's quite a lot of boilerplate.. Hopefully, there's a better way:
+
+```cpp
+ns::person p = createPerson();
+json j = p;
+
+auto p2 = j.get<ns::person>();
+assert(p == p2);
+```
+
+#### Basic usage
+
+To make this work with one of your types, you only need to provide two methods:
+
+```cpp
+using nlohmann::json;
+
+namespace ns {
+void to_json(json& j, person const& p)
+{
+  j = json{{"name", p.name}, {"address", p.address}, {"age", p.age}};
+}
+
+void from_json(json const& j, person& p)
+{
+  p.name = j["name"].get<std::string>();
+  p.address = j["address"].get<std::string>();
+  p.age = j["age"].get<int>();
+}
+} // namespace ns
+```
+
+That's all. When calling the json constructor with your type, your custom `to_json` method will be automatically called.
+Likewise, when calling `get<your_type>()`, the `from_json` method will be called.
+
+Some important things:
+
+* Those methods **MUST** be in your type's namespace, or the library will not be able to locate them (in this example, they are in namespace `ns`, where `person` is defined).
```

Comment:
Is it possible to define a user type without namespace? If so: please mention this. If not, please describe why.